### DISCUSSION TITLE
Why D?

### DISCUSSION THREAD

**Discussion by ww2406** [abraunegg/onedrive#2036] [General]
Hey @abraunegg!

Out of curiosity -- what led you to write this in D over some other language? (Not here to flame or anything, genuinely curious -- had gotten curious about the language awhile back but then realized I hadn't seen anything written in it, and this is the first and it's rather complex and successful!)

**Comment by abraunegg** [comment]
@ww2406 
> Out of curiosity -- what led you to write this in D over some other language?

Unfortunately I was not the original author of the initial work (https://github.com/skilion/onedrive) - so that question is probably best answered by the original author, however their thoughts on D as a language and why it was chosen can be found here: https://aradaelli.com/blog/why-i-like-d/

The question I can answer however is - why, when looking to fix all the known issues / bugs & feature requests folk had - did I not take that opportunity to rewrite the client in Go or C or C++ or something else .. well being the curious mind and technologist that I am with my day-to-day job I saw this as an opportunity to learn something new .. which I have done over the last few years in maintaining and developing this client.

Hope this helps answer your question.

**Reply by ww2406** [reply]
Gotcha! Thanks for the background and for sharing -- love all the work you've put into this.
